lib/widgets/post_action_button.dart

```diff
@@ -35,7 +36,14 @@ class PostSongDetails extends StatelessWidget {
             PostActionButton(
               icon: Icons.play_arrow,
               text: "Listen",
-              onPressed: () {},
+              onPressed: () async {
+                const spotifyUrl =
+                    'https://open.spotify.com/playlist/37i9dQZF1DWX5ZOsG2Ogi1?si=41c6a392cf7d4a2b';
+                final Uri url = Uri.parse(spotifyUrl);
+                if (await canLaunchUrl(url)) {
+                  launchUrl(url, mode: LaunchMode.externalApplication);
+                }
```

Comment:
Can we add a fallback for when `canLaunchUrl` returns false? Perhaps opening the URL in a browser?